File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
	
    public static void main(String[] args) throws IOException {
    	
    	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    	StringTokenizer st;
    	StringBuilder sb = new StringBuilder();
    	
    	int N = Integer.parseInt(br.readLine());
    	boolean[] arr = new boolean[N+1];
    	st = new StringTokenizer(br.readLine());
    	for (int i=1; i<=N; i++) {
    		if (Integer.parseInt(st.nextToken()) == 1) {
    			arr[i] = true;
    		}	
    	}
    	
    	int M = Integer.parseInt(br.readLine());
    	for (int m=0; m<M; m++) {
    		st = new StringTokenizer(br.readLine());
    		int s = Integer.parseInt(st.nextToken());
    		int i = Integer.parseInt(st.nextToken());
    		if (s == 1) {
    			for (int j=i; j<=N; j+=i) {
    				arr[j] = !arr[j];
    			}
    		} else {
    			int d = 0;
    			while (i-d>=1 && i+d<=N && arr[i-d] == arr[i+d]) {
    				d++;
    			}
    			d--;
    			for (int j=i-d; j<=i+d; j++) {
    				arr[j] = !arr[j];
    			}
    		}
    	}
    	
    	for (int i=1; i<=N; i++) {
    		if (arr[i]) {
    			sb.append(1);
    		} else {
    			sb.append(0);
    		}
    		sb.append(" ");
    		if (i%20 == 0) {
    			sb.append("\n");
    		}
    	}
    	
    	System.out.println(sb.toString());
    	
    	
    }
    

        
}

File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.py
import sys

N = int(sys.stdin.readline())
arr = [False] + [bool(int(x)) for x in sys.stdin.readline().split()]

M = int(sys.stdin.readline())
for _ in range(M) :
    s, n = map(int, sys.stdin.readline().split())
    if s == 1 :
        for i in range(n, N+1, n) :
            arr[i] = not arr[i]
    else :
        i = 0
        while n-i >= 1 and n+i <= N and arr[n-i] == arr[n+i] :
            i += 1
        i -= 1
        for j in range(n-i, n+i+1) :
            arr[j] = not arr[j]
m = N//20
i = 0
for i in range(1, N+1) :
    print(int(arr[i]), end=" ")
    if i%20 == 0 :
        print()